R/parallel.R

```diff
@@ -2,23 +2,31 @@
 # Helpers for parallel processing
 
 # object should be a workflow
-get_operator <- function(allow = TRUE, object) {
+allow_parallelism <- function(allow = TRUE, object = NULL) {
   is_par <- foreach::getDoParWorkers() > 1
-  pkgs <- required_pkgs(object)
-  blacklist <- c("keras", "rJava")
-  if (is_par & allow && any(pkgs %in% blacklist)) {
-    pkgs <- pkgs[pkgs %in% blacklist]
-    msg <- paste0("'", pkgs, "'", collapse = ", ")
-    msg <- paste("Some required packages prohibit parallel processing: ", msg)
-    cli::cli_alert_warning(msg)
-    allow <- FALSE
+  if (!is.null(object)) {
+    pkgs <- required_pkgs(object)
+    blacklist <- c("keras", "rJava")
+    if (is_par & allow && any(pkgs %in% blacklist)) {
+      pkgs <- pkgs[pkgs %in% blacklist]
+      msg <- paste0("'", pkgs, "'", collapse = ", ")
+      msg <- paste("Some required packages prohibit parallel processing: ", msg)
+      cli::cli_alert_warning(msg)
+      allow <- FALSE
+    }
   }
 
-  cond <- allow && is_par
+  allow && is_par
+}
+
+get_operator <- function(allow = TRUE, object) {
+  cond <- allow_parallelism(allow, object)
+
   if (cond) {
```

Comment:
Consider the following:

```suggestion
  if (allow_parallelism(allow, object)) {
```

I feel like it reads pretty well

Reply:
I'm for it!